students/EricAdams/session07/test_html_render.py

```diff
@@ -0,0 +1,165 @@
+    # NOTE: you could comment out this if you want to see the file.
+    if remove:
+        os.remove(filename)
+    return contents
+
+
+def test_new_element():
+    el_object = Element()
+    el_object2 = Element('content')
+
+
+def test_add_content():
+    el_object = Element('content')
+    el_object = Element()
+    assert el_object.content == []
+
+
+def test_adding_empty_string():
+    el_object = Element('')
+    assert el_object.content == ['']
+
+
+def test_append_string():
+    el_object = Element('spam, spam, spam')
+    el_object.append(', wonderful spam')
+    assert el_object.content == ['spam, spam, spam', ', wonderful spam']
+
+
+def test_tag_exists():
+    assert Element.tag == 'html'
+    el_object = Element('spam, spam, spam')
+    assert el_object.tag == 'html'
+
+
+def test_indent_exists():
+    assert Element.indent == '  '
+
+
+def test_render():
+    my_stuff = 'spam, spam, spam'
+    el_object = Element(my_stuff)
+    more_stuff = 'eggs, eggs, eggs'
+    el_object.append(more_stuff)
+    contents = render_element(el_object)
+    assert contents.startswith('<html>')
+    assert contents.endswith('</html>')
+    assert my_stuff in contents
+    assert more_stuff in contents
+
+
+# you want to be careful with these:
+# It is testing an implementation detail, which is less than idea.
+#  sometimes in TDD, it's helpful to have quickies tests of
+#  implementation details so you can see that partially written code
+#  is working -- but if/when you can test actual functionality, that's
+#  better. In this case, once we have a render() method, we can test
+#  that the tag gets rendered properly, so don't need to test if the
+#  tag attribute is correct.
+
+# def test_body_tag():
+#    assert Body.tag == 'body'
+
+# def test_p_tag():
+#     assert Para.tag == 'p'
+
+
+# def test_html_tag():
+#     assert HTML.tag == 'html'
+
+def test_render_body():
+    my_stuff = 'spam, spam, spam'
+    el_object = Body(my_stuff)
+    more_stuff = 'eggs, eggs, eggs'
+    el_object.append(more_stuff)
+    contents = render_element(el_object)
+    assert contents.startswith('<body>')
+    assert contents.endswith('</body>')
+    assert my_stuff in contents
+    assert more_stuff in contents
+
+
+def test_render_para():
+    my_stuff = 'spam, spam, spam'
+    el_object = P(my_stuff)
+    more_stuff = 'eggs, eggs, eggs'
+    el_object.append(more_stuff)
+    contents = render_element(el_object)
+    assert contents.startswith('<p>')
+    assert contents.endswith('</p>')
+    assert my_stuff in contents
+    assert more_stuff in contents
+
+
+def test_render_html():
+    my_stuff = 'spam, spam, spam'
+    el_object = Html(my_stuff)
+    more_stuff = 'eggs, eggs, eggs'
+    el_object.append(more_stuff)
+    contents = render_element(el_object)
+    assert contents.startswith('<html>')
+    assert contents.endswith('</html>')
+    assert my_stuff in contents
+    assert more_stuff in contents
+
+
+def test_render_non_strings():
```

Comment:
I see you have my updated tests now -- so this is failing, yes?

I"ll merge and look forward to seeing where this goes.